src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonParameterInfoOfT.cs

```diff
@@ -55,8 +55,7 @@ public override bool ReadJson(ref ReadStack state, ref Utf8JsonReader reader, ou
             bool success;
             bool isNullToken = reader.TokenType == JsonTokenType.Null;
 
-            if (isNullToken &&
-                ((!_converter.HandleNullValue && !state.IsContinuation) || Options.IgnoreNullValues))
+            if (isNullToken && !_converter.HandleNullValue && !state.IsContinuation)
```

Comment:
The change here and on line 87 ensures we have consistent behavior with property deserialization, where we likely throw if `null` is to be assigned to a value type. The previous logic can be reinstated when https://github.com/dotnet/runtime/issues/30795 is addressed.